### DISCUSSION TITLE
How to restart Krohnkite after upgrading for changes to take effect?

### DISCUSSION THREAD

**Discussion by piotr-dobrogost** [anametologin/krohnkite#90] [General]
It looks like after upgrading Krohnkite with `kpackagetool6 -t KWin/Script -u krohnkite.kwinscript` the script should be somehow reloaded for changes to take effect. How can this be done? It would be useful to add this information to README.

Related:
[How to properly reload a kwin script](https://discuss.kde.org/t/how-to-properly-reload-a-kwin-script/4260)
[How to apply KDE Plasma KWin script change without restarting KWin? ](https://discuss.kde.org/t/how-to-apply-kde-plasma-kwin-script-change-without-restarting-kwin/33188)
[How to apply KDE Plasma KWin script change without restarting KWin?](https://stackoverflow.com/q/79440642/95735)

**Comment by rkunschke** [comment]
The easyest way ist the gui-way throw the kwin Scripts settings page.
Just disable the script, press apply, enable it and press apply again.

**Comment by jjspierx** [comment]
Here is a simple shell script that will toggle Kronhkite on/off. 

```
#!/usr/bin/env bash

current=`kreadconfig6 --file kwinrc --group Plugins --key krohnkiteEnabled`
if [ $current = "true" ]; then
  kwriteconfig6 --file kwinrc --group Plugins --key krohnkiteEnabled false
  echo "Krohnkite disabled"
elif [ $current = "false" ]; then
  kwriteconfig6 --file kwinrc --group Plugins --key krohnkiteEnabled true
  echo "Krohnkite enabled"
fi
qdbus6 org.kde.KWin /KWin reconfigure
```

**Reply by piotr-dobrogost** [reply]
Nice. How did you install `qdbus6`?
Recently, when I wanted to install it on Fedora I couldn't find a package providing this tool and I remember reading many complaints that there is a mess regarding packaging this binary.

**Reply by jjspierx** [reply]
I did not explicitly install qdbus6.  For Arch it is a part of `qt6-tools` package.  I think the commands are the same for qdbus/qdbus5 if you are running earlier versions.

**Reply by piotr-dobrogost** [reply]
For those wanting to install Qt 6 version of `qdbus` on Fedora – https://discuss.kde.org/t/what-package-provides-qt6-version-of-qdbus-command-in-fedora/25456?u=piotr_dobrogost

**Reply by piotr-dobrogost** [reply]
@jjspierx 
> Here is a simple shell script that will toggle Kronhkite on/off.

It seems that after toggling Krohnkite off this way it's still active.
Does it work for you?

**Reply by jjspierx** [reply]
It does, I use it all the time.  I created a KWIN shortcut for it, Meta + K toggles it.  I use it mostly when something funny happens and a window won't tile.  Sometimes when switching between Monocle and Tile layouts, a windows gets "stuck".  Hitting Meta+K twice toggles Krohnkite off and then back on, and all my windows are tiled.

**Reply by piotr-dobrogost** [reply]
> Hitting Meta+K twice toggles Krohnkite off and then back on (…)

So it can be said you are in practice using it only to _restart_ Krohnkite not to turn it off.
Try instead pressing it once, play with windows and see that they are still being tiled.

**Reply by jjspierx** [reply]
Are you sure they are still being tiled, and KDE isn't just remembering their last tiled position so it appears they are being tiled?  I just tried it out, and it is definitely not tiling windows as I open them when I press it once to turn it off.

Are you running Plasma 6.x?

**Reply by piotr-dobrogost** [reply]
Yes, tiling is still active. Although it seems it's active for windows which are present when turning off Krohnkite. After quitting some app and starting it again the window is no longer being tiled. I guess some slots of existing windows are still connected to some signals after disabling the script? 

```
[miner@hostx ~]$ kinfo
Operating System: Fedora Linux 42
KDE Plasma Version: 6.3.4
KDE Frameworks Version: 6.12.0
Qt Version: 6.8.2
Kernel Version: 6.14.0-63.fc42.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 4 × Intel® Core™ i3-8100 CPU @ 3.60GHz
Memory: 3.5 GiB of RAM
Graphics Processor 1: Intel® UHD Graphics 630
```

**Reply by piotr-dobrogost** [reply]
Based on https://discuss.kde.org/t/how-to-apply-kde-plasma-kwin-script-change-without-restarting-kwin/33188/12?u=piotr_dobrogost this might work
```
$ qdbus-qt6 org.kde.KWin /Scripting unloadScript "krohnkite"
$ qdbus-qt6 org.kde.KWin /Scripting start
```
although the remark

> # (KWin doesn't reload files, and keep running old versions.)

at https://github.com/anametologin/krohnkite/blob/v0.9.1/bin/load-script.sh#L54 suggests it might not be enough…
